### PR TITLE
Base of the DstController implemented ;fixes #18

### LINKED ISSUE #18: Implement the DstController
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the DEHP-MATLAB
- [x] I have searched [open](https://github.com/RHEAGROUP/DEHP-MATLAB/issues) and [closed](https://github.com/RHEAGROUP/DEHP-MATLAB/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

- [x]  Implement the base DstController with the ability to communicate with the Matlab Application
### System Configuration

- DEHPCommon version:
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-MATLAB/pulls) open
- [x] I have verified that I am following the DEHP-MATLAB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-MATLAB/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #18 
- Connection to Matlab via COM Interop
- Disconnection from Matlab

### CODE REVIEW

**Comment by antoineatrhea** [comment]
image

File: DEHPMatlab/ViewModel/DataSourceViewModel.cs [hunk 1]
```diff
@@ -95,6 +103,8 @@ public string ConnectButtonText
         /// </summary>
         public IHubBrowserHeaderViewModel HubBrowserHeader { get; set; }
 
+        public IStatusBarControlViewModel StatusBar { get; set; }
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/DataSourceViewModel.cs:106]: Missing docs

File: DEHPMatlab/ViewModel/MainWindowViewModel.cs [hunk 2]
```diff
@@ -55,18 +55,23 @@ public class MainWindowViewModel : IMainWindowViewModel
         /// </summary>
         public IStatusBarControlViewModel StatusBarControlViewModel { get; }
 
+        public IDstDataSourceViewModel DstDataSourceViewModel { get; }
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/MainWindowViewModel.cs:58]: Missing docs

File: DEHPMatlab/ViewModel/Row/RowViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace DEHPMatlab.ViewModel.Row
+{
+    using ReactiveUI;
+
+    /// <summary>
+    /// The <see cref="RowViewModel"/> stores the value of variable from the Matlab Workspace
+    /// </summary>
+    public class RowViewModel: ReactiveObject
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/Row/RowViewModel.cs:32]: Maybe a bit more specific classname like "MatlabWorspaceValueRowViewModel"? Very generic right now.

File: DEHPMatlab/ViewModel/Row/RowViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        public string Name
+        {
+            get => this.name;
+            set => this.RaiseAndSetIfChanged(ref this.name, value);
+        }
+
+        /// <summary>
+        /// Backing field for <see cref="Value"/>
+        /// </summary>
+        private object value;
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/Row/RowViewModel.cs:51]: We typically group all privates on top of the file, then constructors then the public props.

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Several small changes requested.

File: DEHPMatlab/DstController/DstController.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        /// </summary>
+        private void InitializeObservables()
+        {
+            this.WhenAnyValue(x => x.matlabConnector.MatlabConnectorStatus)
+                .Subscribe(this.WhenMatlabConnectionStatusChange);
+        }
+
+        /// <summary>
+        /// Update the <see cref="IsSessionOpen"/> when the Matlab connection status changes
+        /// </summary>
```
Comment (nathanatrhea) [inline DEHPMatlab/DstController/DstController.cs:84]: missing doc for the param

File: DEHPMatlab/Services/MatlabConnector/MatlabConnector.cs [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+                this.logger.Error($"Exception: {ex.Message}");
+            }
+
+            this.MatlabConnectorStatus = MatlabConnectorStatus.Disconnected;
+        }
+
+        /// <summary>
+        /// Retrieve a variable from the Matlab workspace
+        /// </summary>
+        /// <param name="variableName">The name of the varible</param>
```
Comment (nathanatrhea) [inline DEHPMatlab/Services/MatlabConnector/MatlabConnector.cs:133]: misspelled *varible

File: DEHPMatlab/ViewModel/DataSourceViewModel.cs [hunk 7]
```diff
@@ -47,6 +47,8 @@ public abstract class DataSourceViewModel : ReactiveObject
         /// </summary>
         private const string DisconnectText = "Disconnect";
 
+        protected string DataSourceName;
```
Comment (nathanatrhea) [inline DEHPMatlab/ViewModel/DataSourceViewModel.cs:50]: missing doc

File: DEHPMatlab/ViewModel/DataSourceViewModel.cs [hunk 8]
```diff
@@ -117,5 +130,20 @@ protected void UpdateConnectButtonText(bool isSessionOpen)
         {
             this.ConnectButtonText = isSessionOpen ? DisconnectText : ConnectText;
         }
+
+        /// <summary>
+        /// Append the connection status to the status bar
+        /// </summary>
+        protected void UpdateStatusBar(bool isSessionOpen)
```
Comment (nathanatrhea) [inline DEHPMatlab/ViewModel/DataSourceViewModel.cs:137]: missing doc

File: DEHPMatlab/ViewModel/MainWindowViewModel.cs [hunk 9]
```diff
@@ -55,18 +55,26 @@ public class MainWindowViewModel : IMainWindowViewModel
...
+            IHubDataSourceViewModel hubDataSourceViewModel, IStatusBarControlViewModel statusBarControlViewModel
+            ,IDstDataSourceViewModel dstDataSourceViewModel)
         {
             this.navigationService = navigationService;
             this.HubDataSourceViewModel = hubDataSourceViewModel;
             this.StatusBarControlViewModel = statusBarControlViewModel;
+            this.DstDataSourceViewModel = dstDataSourceViewModel;
         }
     }
 }
```
Comment (nathanatrhea) [inline DEHPMatlab/ViewModel/MainWindowViewModel.cs:80]: missing end of file empty line

**Review by nathanatrhea — CHANGES_REQUESTED** [review]
Doc changes requested

**Review by alexatrhea — APPROVED** [review]

**Review by nathanatrhea — APPROVED** [review]

**Pull request merged**